aspnetcore/tutorials/razor-pages/sql/includes/sql3.md

```diff
@@ -28,7 +28,9 @@ The generated connection string will be similar to the following:
 
 ---
 
-When the app is deployed to a test or production server, an environment variable can be used to set the connection string to a test or production database server. See [Configuration](xref:fundamentals/configuration/index) for more information.
+When the app is deployed to a test, non-production server, an environment variable can be used to set the connection string to a test, non-production database server. For more information, see [Configuration](xref:fundamentals/configuration/index).
```

Comment:
```suggestion
When the app is deployed to a test server, an environment variable can be used to set the connection string to a test database server. For more information, see [Configuration](xref:fundamentals/configuration/index). An environment variable should not be used in a production system.
```

@tdykstra I think that's easier to read.

Reply:
@tdykstra  I think we should put the above in an include file and use it everywhere. Then we only have one place to change it. What do you think?

We'll need this boilerplate in MVC and other place. We could add it to EF tutorials.

Reply:
@wadepickett  @tdykstra  we need to work in something like this.

For VS/local db:
In this tutorial, localdb is used in dev on the local machine and doesn't use a password, therefore, a secure authentication flow is not required. 

Then something similar for SqlLite

Reply:
We don't want to spook them right away so we need to tell them up front that since this tutorial uses a local db that doesn't use a password, a secure authentication flow isn't required. Maybe tell them that before the big warning.

Reply:
Sure we can create another include for that case too.  Where you thinking one for VS and one for VSC/VS Windows combined?

Clarification/question: For VS/locaDB, when Trusted_Connection=True as we are using in the connect here, that results in Windows Authentication used. Wouldn't we consider that a type of Authentication flow?  So to say one isn't required might be confusing.   It may not be the best one for the situation depending, but is one.  Maybe we are keeping to a narrower definition of Authenticaiton flow?

Reply:
> @Rick-Anderson, sure we can create another include for that case too only for VS/localDB
> 
> Clarification/question: For VS/localDB, when Trusted_Connection=True as we are using in the connect here, that results in Windows Authentication used. Wouldn't we consider that a type of Authentication flow? So to say one isn't required might be confusing. It may not be the best one for the situation depending, but is one. Maybe we are keeping to a narrower definition of Authentication flow?

 Resource Owner Password Credentials (ROPC) grant is using a PW. No PW is used with local DB. We're only concerned where a  PW is used, like the old deployment method where we put the PW in an env variable.

Reply:
>@tdykstra I think that's easier to read.

Yes, and I tweaked it a bit.

>@tdykstra I think we should put the above in an include file and use it everywhere. Then we only have one place to change it. What do you think?

Sounds good, but I don't think we should say "don't use environment variables in production" without saying what is OK to use and providing a link to how-to instructions.

Reply:
> "Then something similar for SqlLite"

@Rick-Anderson, I think the way you worded it in the last edit already fits both situations nicely by specifying "a local database" rather than more specifically "localDB". 

> "In this tutorial, a local database that doesn't need a password is used."

So only the _one_ include would be needed for the note on using a local db.  I will just add one include with your text, unless you still want some sort of separate message for SQLite.

Reply:
> Sounds good, but I don't think we should say "don't use environment variables in production" without saying what is OK to use and providing a link to how-to instructions.
@tdykstra, to address this I added "An environment variable should not be used in a production system. For a production system, use a secure method such as [Azure Key Vault configuration provider](xref:security/key-vault-configuration) or [Managed Identities for Azure resources](/sql/connect/ado-net/sql/azure-active-directory-authentication#using-managed-identity-authentication) to manage sensitive information like connection strings."

See include.  If they are not using Azure SQL server, then it would have to be  key vault I assume, if they are then it would be Managed Identities for Azure.

However, between both includes it feels a little duplicated so maybe I need to address it differently in a combined way.  See what you think.

Reply:
> However, between both includes it feels a little duplicated so maybe I need to address it differently in a combined way. See what you think.